src/test/scala/ai/privado/languageEngine/go/tagger/collection/CollectionTaggerTest.scala

```diff
@@ -91,3 +91,198 @@ class CollectionRestTaggerTest extends CollectionTaggerTestBase {
+        | e.Use(middleware.Logger())
+        |
+        | e.GET("/articles", allArticles)
+        |}
+        |
+        |func allArticles() {
+        | firstName := "anything"
+        |	fmt.Println(firstName)
+        |}
+        |""".stripMargin
+    )
+
+    testClassMap.toMap
+  }
+
+  "CollectionTagger" should {
+    "Test correct collection tagging" in {
+      cpg.method("allArticles").tag.nameExact("COLLECTION_METHOD_ENDPOINT").head.value shouldBe "\"/articles\""
+    }
+  }
+}
+
+class CollectionTaggerChiFrameworkTest extends CollectionTaggerTestBase {
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    new CollectionTagger(cpg, ruleCache).createAndApply()
+  }
+
+  override val goFileContentMap: Map[String, String] = getContent()
+
+  private def getContent(): Map[String, String] = {
+    val testClassMap = mutable.Map[String, String]()
+
+    testClassMap.put(
+      "routes",
+      """
+        |package main
+        |
+        |import (
+        | "fmt"
+        | "github.com/go-chi/chi"
+        |)
+        |
+        |func fetchArticle() {
+        | router := app.router
+        | router.Group(func(r chi.Router) {
+        | r.Use(app.logger.NewRequestLogger())
+        |		router.Get("/articles", allArticles)
+        | })
+        |}
+        |
+        |func allArticles() {
+        | firstName := "anything"
+        |	fmt.Println(firstName)
+        |}
+        |""".stripMargin
+    )
+
+    testClassMap.toMap
+  }
+
+  // ast.FuncLit AST type is not handled yet.
+  "CollectionTagger" ignore {
+    "Test correct collection tagging" in {
+      cpg.method("allArticles").tag.nameExact("COLLECTION_METHOD_ENDPOINT").head.value shouldBe "\"/articles\""
+    }
+  }
+}
+
+class CollectionTaggerGoFiberFrameworkTest extends CollectionTaggerTestBase {
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    new CollectionTagger(cpg, ruleCache).createAndApply()
+  }
+
+  override val goFileContentMap: Map[String, String] = getContent()
+
+  private def getContent(): Map[String, String] = {
+    val testClassMap = mutable.Map[String, String]()
+
+    testClassMap.put(
+      "routes",
+      """
+        |package main
+        |
+        |import (
+        | "fmt"
+        | "github.com/NikSchaefer/go-fiber/middleware"
+        | "github.com/gofiber/fiber/v2"
+        |)
+        |
+        |func fetchArticle(router *fiber.App) {
+        | router.Use(middleware.Security)
+        | router.Post("/articles", allArticles)
+        |}
+        |
+        |func allArticles() {
+        | firstName := "anything"
+        |	fmt.Println(firstName)
+        |}
+        |""".stripMargin
+    )
+
+    testClassMap.toMap
+  }
+
+  "CollectionTagger" should {
+    "Test correct collection tagging" in {
+      cpg.method("allArticles").tag.nameExact("COLLECTION_METHOD_ENDPOINT").head.value shouldBe "\"/articles\""
+    }
+  }
+}
+
+class CollectionTaggerGoBuffaloFrameworkTest extends CollectionTaggerTestBase {
```

Comment:
@ankit-privado Why we have created a class for each unit test?

Reply:
@pandurangpatil For every ORM framework, I have created its own test class.

Reply:
I think we can merge them into one class. But looks like the `private def getContent()` and `goFileContentMap` variables need to be defined separately. 

For now I am approving it. We have to work on a proper solution and merge all of these unit tests inside one class.